include/unifex/let_with.hpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2019-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/just.hpp>
+#include <unifex/let.hpp>
+#include <unifex/receiver_concepts.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/execution_policy.hpp>
+#include <unifex/type_list.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+
+namespace _let_with {
+
+template<typename StateFactory, typename InnerOp, typename Receiver>
+struct _let_with_operation {
+  struct type;
+};
+
+template <typename StateFactory, typename InnerOp, typename Receiver>
+using operation =  typename _let_with_operation<
+    StateFactory, InnerOp, Receiver>::type;
+
+template<typename StateFactory, typename SuccessorFactory>
+struct _let_with_sender {
+    class type;
+};
+
+template<typename StateFactory, typename SuccessorFactory>
+using let_with_sender = typename _let_with_sender<StateFactory, SuccessorFactory>::type;
+
+template<typename StateFactory, typename SuccessorFactory>
+class _let_with_sender<StateFactory, SuccessorFactory>::type {
+public:
+    using InnerOp = std::invoke_result_t<SuccessorFactory, std::invoke_result_t<StateFactory>&>;
```

Comment:
```suggestion
    using InnerOp = std::invoke_result_t<SuccessorFactory, callable_result_t<StateFactory>&>;
```

... since a nullary invocable must be a simple callable (not a member pointer).